src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -366,9 +373,6 @@ public boolean apply(@Nullable Container container) {
      *
      */
     private synchronized boolean addProvisionedSlave(String ami, int amiCap) throws Exception {
-        if( amiCap == 0 )
-            return true;
-
```

Comment:
When amiCap is zero, this if statement causes the containerCap (server maximum) to not be checked. This means an infinite number of containers could be spawned even when the containerCap is set.

Additionaly this return bypasses the provisoningAmi's counting. which means the count of running slaves will probably be wrong (I believe, I don't fully understand the provisioningAmi's object).

If it returned false this would be a totally different story because then nothing would be provisioned.